lib/freeze.js

```diff
@@ -37,42 +34,53 @@ module.exports = class Freeze {
     });
   }
 
-  freeze(context, props) {
+  async freeze(context, props) {
     const labels = context.payload.issue.labels;
     if (GitHubHelper.isLabelOnIssue(
       context.payload.issue,
       this.config.labelName) === undefined) {
       labels.push(this.config.labelName);
     }
 
-    this.github.issues.edit(context.issue({
+    const kv = metadata(context.github, context.issue(), process.env.APP_ID);
+    await kv.set('snooze', props);
+
+    await this.github.issues.edit(context.issue({
       state: 'closed',
       labels
     }));
 
-    this.github.issues.createComment(context.issue({
+    await this.github.issues.createComment(context.issue({
       body: 'Sure thing. I\'ll close this issue for a bit. I\'ll ping you around ' +
-      moment(props.unfreezeMoment).calendar() + ' :clock1: ' +
-      '<!-- ' + JSON.stringify(props) + '-->'
+      moment(props.unfreezeMoment).calendar() + ' :clock1: '
     }));
   }
 
-  unfreeze(issue, props) {
+  async checkUnfreeze(issue) {
+    const kv = metadata(this.github, issue, process.env.APP_ID);
+    const props = await kv.get('snooze');
+
+    if (moment(props.unfreezeMoment) < moment()) {
+      return this.unfreeze(issue, props);
+    }
+  }
+
+  async unfreeze(issue, props) {
     const labels = issue.labels;
     const frozenLabel = labels.find(label => {
       return label.name === this.config.labelName;
     });
     const pos = labels.indexOf(frozenLabel);
     labels.splice(pos, 1);
 
-    this.github.issues.edit(Object.assign(GitHubHelper.parseCommentURL(issue.comments_url), {
-      state: 'open',
-      labels
-    }));
+    const {owner, repo, number} = issue;
```

Comment:
what sort of black magic is this?!

Reply:
[destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)

Reply:
that's damn cool.